### PR TITLE
fixed the window bounds + migrated from N3.js to rdf-stores.js

### PR DESCRIPTION
**Pull request by argahsuknesib**

### CODE REVIEW

**Comment by pbonte** [comment]
What are container bounds?

**Comment by argahsuknesib** [comment]
The bounds of the window subset of a stream which contains a particular quad container. I will remane it to window bounds, as it makes more sense.

File: dist/operators/s2r.d.ts [hunk 1]
```diff
@@ -42,5 +42,10 @@ export declare class CSPARQLWindow {
     add(e: Quad, timestamp: number): void;
     compute_report(w: WindowInstance, content: QuadContainer, timestamp: number): boolean;
     scope(t_e: number): void;
-    subscribe(output: 'RStream' | 'IStream' | 'DStream', call_back: (data: QuadContainer) => void): void;
+    subscribe(output: 'RStream' | 'IStream' | 'DStream', call_back: (data: container_with_bounds) => void): void;
 }
+export type container_with_bounds = {
```
Comment (pbonte) [inline dist/operators/s2r.d.ts:47]: Why is container_with_bounds not camel case like the other types?

File: dist/operators/s2r.d.ts [hunk 2]
```diff
@@ -42,5 +42,10 @@ export declare class CSPARQLWindow {
     add(e: Quad, timestamp: number): void;
     compute_report(w: WindowInstance, content: QuadContainer, timestamp: number): boolean;
     scope(t_e: number): void;
-    subscribe(output: 'RStream' | 'IStream' | 'DStream', call_back: (data: QuadContainer) => void): void;
+    subscribe(output: 'RStream' | 'IStream' | 'DStream', call_back: (data: container_with_bounds) => void): void;
```
Comment (pbonte) [inline dist/operators/s2r.d.ts:45]: I would suggest to just pass a metadata object with the QuadContainer in stead, so it is more flexible. Also I dont like the name ContainerWithBounds, when picking a name for a class/function/variable it should be clear from the name what it describes/does/contains...

File: dist/operators/s2r.d.ts (hunk 1 above)
Comment (pbonte) [reply dist/operators/s2r.d.ts:47]: I'm also not sure why you need it? The QuadContainer already has the timestamp the window triggered,i.e. last_time_stamp_changed.

**Review by pbonte — COMMENTED** [review]
Hi @argahsuknesib , thanks for submitting!
Some small comments on the PR

**Pull request closed without merging**